test/built-ins/TypedArrays/ctors-bigint/object-arg/string-tobigint.js

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2018 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-typedarray-object
+description: >
+  Behavoir for input array of Strings, successful conversion
+info: |
+  22.2.4.4 TypedArray ( object )
+
+  This description applies only if the TypedArray function is called with at
+  least one argument and the Type of the first argument is Object and that
+  object does not have either a [[TypedArrayName]] or an [[ArrayBufferData]]
+  internal slot.
+
+  ...
+  8. Repeat, while k < len
+    ...
+    b. Let kValue be ? Get(arrayLike, Pk).
+    c. Perform ? Set(O, Pk, kValue, true).
+  ...
+
+  9.4.5.5 [[Set]] ( P, V, Receiver)
+
+  ...
+  2. If Type(P) is String and if SameValue(O, Receiver) is true, then
+    a. Let numericIndex be ! CanonicalNumericIndexString(P).
+    b. If numericIndex is not undefined, then
+      i. Return ? IntegerIndexedElementSet(O, numericIndex, V).
+  ...
+
+  9.4.5.9 IntegerIndexedElementSet ( O, index, value )
+
+  ...
+  5. If arrayTypeName is "BigUint64Array" or "BigInt64Array",
+     let numValue be ? ToBigInt(value).
+  ...
+
+  ToBigInt ( argument )
+
+  Object, Apply the following steps:
+    1. Let prim be ? ToPrimitive(argument, hint Number).
+    2. Return the value that prim corresponds to in Table [BigInt Conversions]
+
+  BigInt Conversions
+    Argument Type: String
+    Result:
+      1. Let n be StringToBigInt(prim).
+      2. If n is NaN, throw a SyntaxError exception.
+      3. Return n.
+
+includes: [testBigIntTypedArray.js]
+features: [BigInt, TypedArray]
+---*/
+
+testWithBigIntTypedArrayConstructors(function(TA) {
+  var typedArray = new TA(['', '1n']);
```

Comment:
a nit pick: '1n' is invalid and this should throw a SyntaxError.

Trick: basically if any string `x` returns `NaN` for `Number(x)`, then `BigInt(x)` should throw a SyntaxError.

More info [here](https://tc39.github.io/proposal-bigint/#sec-string-to-bigint) and [here](https://tc39.github.io/proposal-bigint/#sec-tonumber-applied-to-the-string-type):

`A StringNumericLiteral may not include a BigIntLiteralSuffix.`

So this would work nicely with `'1'`, `'42'`, etc. but not anything with the `n` suffix.

We should have a test for the SyntaxError when these other strings. Also: https://github.com/tc39/proposal-bigint/pull/130